eng/common/pipelines/templates/steps/eng-common-workflow-enforcer.yml

```diff
@@ -1,18 +1,18 @@
+parameters:
+  ScriptDirectory: 'eng/common/scripts/git-diff-changes.ps1'
 steps:
   - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
     - pwsh: |
         Set-PsDebug -Trace 1
         # Find the default branch of the repo
-        $defaultBranch = (git remote show origin | Out-String) -replace "(?ms).*HEAD branch: (\w+).*", '$1'
-        $targetBranch = "$(System.PullRequest.TargetBranch)" -replace "refs/heads/"
-
-        Write-Host "Default Branch: ${defaultBranch}"
-
+        Write-Host "Default Branch: $(defaultBranch)"
```

Comment:
Lets change this to use https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/pipelines/templates/steps/set-default-branch.yml.

Reply:
The place using this template has already added the set default branch step. I will add condition on this step to avoid same call.